spec/lib/lorekeeper/json_logger_spec.rb

```diff
@@ -286,6 +297,29 @@
           logger.write(message)
           expect(io.received_message).to eq(message)
         end
+
+        context 'non-representable data' do
+          let(:message) { { message: Float::NAN } }
+          let(:message_that_raises) { { message: 'I am raising an error' } }
+          let(:error_message) { 'Houston, we have a problem!' }
+
+          it 'falls back to :object mode if it can' do
+            logger.write(message)
+            # it's non NAN anymore since we're adding a new line to each message
+            #
+            expect(io.received_message).to eq({ message: Float::INFINITY })
+          end
+
+          it 'serializes error message in case of raising an exception' do
+            allow(Oj).to receive(:dump)
+              .with(message_that_raises, { cache_keys: true, cache_str: 5, mode: :compat })
+              .and_raise(error_message)
```

Comment:
could we instead find an input that in fact would raise this error?

Reply:
@jcarres-mdsol 

hmm, I looked at the `Oj.dump` supported data types table that I mentioned in PR description but it seems that it works almost with everything 😓 

I tried `to_json` but it works.
The other one that not suppose to work is `ActiveRecord`

So maybe the question is whether it should even catch an error if I can't come up with a real one?  🤔

Reply:
probably we can utilize `default_options` to raise some error:
```rb
        context 'non-representable data' do
          let(:message) { { message: Float::NAN } }

          it 'falls back to :object mode if it can' do
            Oj.default_options = { mode: :object }
            logger.write(message)
            expect(io.received_message).to eq(message: Float::INFINITY)
          end

          it 'serializes error message in case of raising an exception' do
            Oj.default_options = { mode: :strict }
            logger.write(message)
            expect(io.received_message).to eq("message" => "Failed to dump Float Object to JSON in strict mode.\n")
          end
        end
```

and I noticed the `rescue` clause is not working as expected. we need to make it nested like:
```rb
    def write(message)
      super(json_message(message) << "\n")
    end

    private

    def json_message(message)
      Oj.dump(message, mode: :compat, cache_keys: true, cache_str: 5)
    rescue JSON::GeneratorError
      begin
        Oj.dump(message)
      rescue => e
        Oj.dump(MESSAGE => e.message)
      end
    end
```